heed/src/env.rs

```diff
@@ -210,13 +217,27 @@ unsafe impl Send for EnvInner {}
 
 unsafe impl Sync for EnvInner {}
 
+impl Drop for EnvInner {
+    fn drop(&mut self) {
+        // We lock the environment map to ensure that no one tries to
+        // open this environment at the same time as we are closing it.
+        let _lock = OPENED_ENV.lock().unwrap();
```

Comment:
This will obviously trigger a dead-lock as will already locked the `OPENED_ENV` mutex to be able to drop the last reference of the env, we could use a `parking_lot` mutex as it is reentrant, when two locks are acquiered on the same thread it works fine.

Reply:
Or we can maybe replace the `Env` in the `OPENED_ENV` `HashMap` by a `None` to indicate that it is being dropped right now and acquire the lock again in the `Env::drop` method and remove the entire entry. This way if another thread tries to open the environment when we are trying to `close` it, it must be inform that the env is currently in a closing operation and must not be opened, an error must be returned.